packages/x-data-grid-pro/src/hooks/features/infiniteLoader/useGridInfiniteLoader.tsx

```diff
@@ -35,6 +35,7 @@ export const useGridInfiniteLoader = (
   const visibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
   const currentPage = useGridVisibleRows(apiRef, props);
   const observer = React.useRef<IntersectionObserver>();
+  const updateTargetTimeout = React.useRef<ReturnType<typeof setTimeout>>();
```

Comment:
We have a `useTimeout` utility to deal with [a performance issue](https://github.com/mui/material-ui/pull/39060), it's also more readable and handles cleanup automatically.